public/inetchannel.h

```diff
@@ -57,82 +55,93 @@ enum NetChannelBufType_t : int8
 
-	virtual const netadr_t	&GetRemoteAddress( void ) const = 0;
-	virtual int				GetDropNumber( void ) const = 0;
+	virtual const netadr_t &GetRemoteAddress( void ) const = 0;
 
 	virtual void	UpdateMessageStats( int msggroup, int bits, bool ) = 0;
+	
+	virtual void	unk001( void ) = 0;
+	
 	virtual bool	CanPacket( void ) const = 0;
 	virtual bool	IsOverflowed( void ) const = 0;
 	virtual bool	HasPendingReliableData( void ) = 0;
 
-	virtual void	SetMaxBufferSize( NetChannelBufType_t bufferType, int nBytes ) = 0;
-
 	// For routing messages to a different handler
 	virtual bool	SetActiveChannel( INetChannel *pNewChannel ) = 0;
-	virtual void	AttachSplitPlayer( CSplitScreenPlayerSlot nSplitPlayerSlot, INetChannel *pChannel ) = 0;
-	virtual void	DetachSplitPlayer( CSplitScreenPlayerSlot nSplitPlayerSlot ) = 0;
+	
+	virtual void	AttachSplitPlayer( CSplitScreenSlot nSlot, INetChannel *pChannel ) = 0;
+	virtual void	DetachSplitPlayer( CSplitScreenSlot nSlot ) = 0;
 
-	virtual void	SetUsesMaxRoutablePlayload(bool useMax) = 0;
-	virtual void	SetDataRate(float rate) = 0;
-	virtual void	SetUpdateRate( int rate ) = 0;
-	virtual void	SetCommandRate( int rate ) = 0;
+	virtual void	SetMinDataRate( int rate ) = 0;
+	virtual void	SetMaxDataRate( int rate ) = 0;
 
-	virtual bool	IsTimedOut( void ) const  = 0;
+	virtual void	SetTimeout( float seconds, bool bForceExact = false ) = 0;
+	virtual bool	IsTimedOut( void ) const = 0;
+	virtual void	UpdateLastReceivedTime( void ) = 0;
 
-	virtual void	SetRemoteFramerate( float flFrameTime, float flFrameTimeStdDeviation, float flFrameStartTimeStdDeviation ) = 0;
-	virtual bool	IsRemoteDisconnected() const = 0;
+	virtual void	SetRemoteFramerate( float flFrameTime, float flFrameTimeStdDeviation, float flFrameStartTimeStdDeviation, float flLoss, float flUnfilteredFrameTime ) = 0;					
+	virtual bool	IsRemoteDisconnected( ENetworkDisconnectionReason &reason ) const = 0;
 
 	virtual void	SetNetMessageDispatcher( INetMessageDispatcher *pDispatcher ) = 0;
 	virtual INetMessageDispatcher *GetNetMessageDispatcher( void ) const = 0;
-	virtual void	SendNetMessage( NetMessageHandle_t msg, const void *pData, NetChannelBufType_t bufType ) = 0;
+	
 	virtual void	StartRegisteringMessageHandlers( void ) = 0;
 	virtual void	FinishRegisteringMessageHandlers( void ) = 0;
-	virtual void	RegisterNetMessageHandlerAbstract( CUtlSlot *, const CUtlAbstractDelegate &, int, NetMessageHandle_t , int ) = 0;
-	virtual void	UnregisterNetMessageHandlerAbstract( CUtlSlot *, const CUtlAbstractDelegate &, NetMessageHandle_t ) = 0;
+	
+	virtual void	RegisterNetMessageHandlerAbstract( CUtlSlot *nSlot, const CUtlAbstractDelegate &delegate, int nParamCount, INetworkSerializable *pNetMessage, int nPriority ) = 0;
+	virtual void	UnregisterNetMessageHandlerAbstract( CUtlSlot *nSlot, const CUtlAbstractDelegate &delegate, INetworkSerializable *pNetMessage ) = 0;
+	
 	virtual void	SetChallengeNr( unsigned int challenge ) = 0;

```

Comment:
In the ``INetworkGameServer`` challenge was changed to a string, can you verify what this method is actually using?

Reply:
It looks like it is being passed as an integer

```CNetChan::SetChallengeNr```:
![img1](https://github.com/alliedmodders/hl2sdk/assets/152704532/95b8e2b2-4102-4328-9cab-a56ae105472a)

```CNetworkGameServerBase::ConnectClient```:
![img2](https://github.com/alliedmodders/hl2sdk/assets/152704532/a1c51c9d-5a13-474b-b7d8-8c9e9ec73660)

I assume that it takes this value from C2S_CONNECT
https://github.com/SteamDatabase/GameTracking-CS2/blob/master/Protobufs/connectionless_netmessages.proto#L6

Reply:
Hm, interesting, then all good